packages/ec-core/src/operator/selector/weighted/weighted_selector_pair.rs

```diff
@@ -0,0 +1,95 @@
+use rand::distr::{Bernoulli, Distribution};
+
+use super::{weighted::Weighted, WeightSumOverflow, WithWeight};
+use crate::{operator::selector::Selector, population::Population};
+
+pub struct WeightedSelectorPair<A, B> {
+    pub(crate) a: A,
+    pub(crate) b: B,
+    pub(crate) distr: Option<Bernoulli>,
+    pub(crate) weight_sum: u32,
+}
+
+#[derive(Debug)]
+pub enum WeightedSelectorsError<A, B> {
+    A(A),
+    B(B),
+
+    ZeroWeightSum,
+
+    // #[error("Adding the weights {0} and {1} overflows")]
+    WeightSumOverflows(u32, u32),
+}
```

Comment:
Depending on where we put this I'd create a errors module or leave it here - also, we need to implement Display, Error, and Diagnostic for this.